### DISCUSSION TITLE
Easy duplication of routes with the new defineRoutes

### DISCUSSION THREAD

**Discussion by depsimon** [remix-run/react-router#11632] [Proposals]
I was reading the [new defineRoute implementation](https://github.com/remix-run/react-router/pull/11596) and was thinking it would be a good idea to think about route duplication during the development.

Would be cool if this new way of defining routes would allow for easy route duplications somehow.

I'd love to be able to do something like this (current API):

```ts
const customRoutes = defineRoutes((route) => {
  route("/", "routes/_app.tsx"); // Go to /issues and you see all issues you have access to
  route("/organization/:organizationId", "routes/_app.tsx"); // Go to /organization/:organizationId/issues and you see all issues for the organization
  route("/organization/:organizationId/project/:projectId", "routes/_app.tsx"); // Go to /organization/:organizationId/project/:projectId/issues and you see all issues for the specific project of the organization
});
```

This currently throws an error:

```
Error: Unable to define routes with duplicate route id: "routes/_app"
```

If we specify unique IDs it prevents the error, but doesn't register the new routes.

Anyway, it'd be great if this scenario was possible easily without having to play with redirections & duplications of real route files.

**Comment by darylhedley** [comment]
@depsimon - I think you can do this already after giving it a try in my project. Have a look at this issue:
https://github.com/remix-run/remix/issues/1898

Looks like in the route file you will get props and on the props you'll find `matches` and within that array you can determine which components to render based upon the third param you pass to `route`

For example, I'm using it like so:
```
route(":id/create", "./modules/scenarios/routes/createScenarioRoute.tsx", { id: "create" }),
route(":id/build", "./modules/scenarios/routes/createScenarioRoute.tsx")
```

I then use the `matches` array and find out if the `id` param exists and if so I render something completely different. Pretty nice if you want to build a whole new view whilst keeping old code in place!
